parsers/CZ.py

```diff
@@ -174,17 +219,28 @@ def fetch_production(
     data_tag = xml.find("data")
     production_breakdowns = ProductionBreakdownList(logger)
 
+    pumping_load_at_time = get_pumping_load(
+        zone_key=zone_key,
+        session=session,
+        target_datetime=target_datetime,
+        logger=logger,
+    )
+
     if data_tag is not None:
         for values in data_tag:
             production = ProductionMix()
             storage = StorageMix()
-
             for k, v in mapper.items():
                 generator = translate_table_gen[k]
                 if k != "PsPP":
                     production.add_value(mode=generator, value=float(values[v]))
                 else:
-                    storage.add_value(mode=generator, value=float(values[v]) * -1)
+                    pumping_storage = float(values[v]) * -1
+                    if values["date"] in pumping_load_at_time:
+                        pumping_storage += pumping_load_at_time[values["date"]]
```

Comment:
Why are you summing things up here?

If there are multiple storage entries they should be looped over and add_value should be called on each of them as it handles summing up values of the same type.